src/cli/generate-idlnames.js

```diff
@@ -0,0 +1,233 @@
+      // properties throughout the structure, and extend it with spec info
+      idl = Object.assign(specInfo(spec), JSON.parse(JSON.stringify(idl)));
+
+      if (names[name]) {
+        // That should never happen, yet it does:
+        // IDL names are sometimes defined in multiple specs. Let's consider
+        // that the "first" (in order of apparence in the report) apparence is
+        // the main one, and let's ignore the second definition.
+        console.warn('[warn] IDL Names - Name defined more than once', name);
+        return;
+      }
+      names[name] = {
+        defined: idl,
+        extended: [],
+        inheritance: idl.inheritance,
+        includes: []
+      };
+    });
+  });
+
+  // Add definitions that extend base definitions
+  results.forEach(spec => {
+    if (!spec.idl || !spec.idl.idlExtendedNames) {
+      return;
+    }
+    Object.entries(spec.idl.idlExtendedNames).forEach(([name, extensions]) =>
+      extensions.forEach(idl => {
+        // Make a deep copy of the IDL block since we're going to add "href"
+        // properties throughout the structure, and extend it with spec info
+        idl = Object.assign(specInfo(spec), JSON.parse(JSON.stringify(idl)));
+
+        if (!names[name]) {
+          // That should never happen, and it does not in practice unless there
+          // was a crawling error on the spec that normally defines the base
+          // IDL name. Alas, such crawling errors do happen from time to time.
+          console.warn('[warn] IDL Names - No definition found', name);
+          names[name] = {
+            extended: [],
+            includes: []
+          };
+        }
+        if (idl.includes) {
+          names[name].includes.push(idl.includes);
+        }
+        names[name].extended.push(idl);
+      }));
+  });
+
+  // Look at extracted dfns and complete the parsed IDL structures with links
+  // to the definitions of the terms in the spec
+  Object.values(names).forEach(desc => {
+    if (desc.defined) {
+      linkifyIdl(desc.defined, results);
+    }
+    if (desc.extended) {
+      desc.extended.forEach(ext => linkifyIdl(ext, results));
+    }
+  });
+
+  // Expand inheritance and includes info
+  Object.values(names).forEach(desc => {
+    if (desc.includes) {
+      desc.includes = desc.includes.map(name => names[name]);
+    }
+    if (desc.inheritance) {
+      desc.inheritance = names[desc.inheritance];
+    }
+  });
+
+  // The expansions were done by reference. In the end, we'll want to serialize
+  // the structure, so we need to make sure that there aren't any cycle. Mixins
+  // cannot create cycles, but inheritance chains can, if not done properly.
+  Object.entries(names).forEach(([name, desc]) => {
+    let current = desc;
+    while (current) {
+      current = current.inheritance;
+      if (current && (current.name === name)) {
+        console.warn('[warn] IDL Names - Cyclic inheritance chain detected', name);
```

Comment:
do we detect that in the anomalies report? (does this actually happen with the current data?)

Reply:
We don't detect that (I created #490) and it does not happen with the current data.
I just felt that I had to take that into account here not to end up with a cryptic serialization failure message if there ever comes a time when a spec creates a cycle.